src/helper/directory_helper.rs

```diff
@@ -64,59 +63,41 @@ impl DirectoryHelper {
                   directory_to_delete: &String) -> Result<(), ::errors::NfsError> {
             let pos = try!(parent_directory.get_sub_directory_index(directory_to_delete).ok_or(::errors::NfsError::DirectoryNotFound)); {
             parent_directory.get_mut_sub_directories().remove(pos);
-            // TODO why is it required to update the grand-parent ? Call update_directory_listing
-            // if no other issue.
-            // TODO Why is lint not working ? The return type is ignored which should have been
-            // caught by lint.
-            try!(self.update_directory_listing_and_parent(parent_directory));
+            parent_directory.get_mut_metadata().set_modified_time(::time::now_utc());
+            try!(self.update_directory_listing(parent_directory));
```

Comment:
The TODO point also mentioned:
```
// TODO Why is lint not working ? The return type is ignored which should have been
// caught by lint.
```
This is still not resolved. Pls check why lint is not complaining.